### ISSUE TITLE
[Slider] Two-Way Binding to Slider.Value produces Stack Overflow

### ISSUE THREAD

**Issue by SuperJMN** [AvaloniaUI/Avalonia#824]
It happened with this Binding:

`<Slider Value="{Binding MyValue, Mode=TwoWay}"  />`

To avoid the problem I had to set the OneWayToSource mode.

**Comment by grokys** [comment]
Could we get a repro for this? I tried changing [this control](https://github.com/AvaloniaUI/Avalonia/blob/master/samples/BindingTest/MainWindow.xaml#L37) to a slider with a two-way binding but didn't get a stack overflow.

**Comment by donandren** [comment]
With the recent versions I'm also experiencing StackOverflow with Binding in some situations. I think this exception is not related to Slider, but it's more general problem in the new binding subsystem (from the last few months). Will try to find repro or unit test.

**Comment by danwalmsley** [comment]
@jameswalmsley also experienced this bug.

**Comment by danwalmsley** [comment]
Just experienced this too!

**Comment by danwalmsley** [comment]
https://files.gitter.im/AvaloniaUI/Avalonia/tqIp/SliderRepro.7z

**Comment by danwalmsley** [comment]
above file is a minimal repro

**Comment by wieslawsoltes** [comment]
Had same issue today.
```XAML
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:local="clr-namespace:Core2D.Avalonia.Views.Style;assembly=Core2D.Avalonia">
    <Grid RowDefinitions="Auto,Auto,Auto,Auto" ColumnDefinitions="Auto,*,50">
        <TextBlock Classes="default" Grid.Column="0" Grid.Row="0" Text="A"/>
        <Slider Classes="default" Grid.Column="1" Grid.Row="0" Value="{Binding A, Mode=TwoWay}" Minimum="0" Maximum="255"/>
        <TextBox Classes="default" Grid.Column="2" Grid.Row="0" Text="{Binding A, Mode=TwoWay}"/>
    </Grid>
</UserControl>
```

**Comment by kekekeks** [comment]
Clicking on a checkbox in BindingTest also produces a stack overflow

**Comment by jkoritzinsky** [comment]
Ok I think I have an understanding of this stack overflow. The checkbox in Binding test is a different one.

**Comment by jkoritzinsky** [comment]
Fixed by #856.
